src/main/java/de/tum/in/www1/artemis/service/ModelingSubmissionService.java

```diff
@@ -39,85 +40,148 @@ public ModelingSubmissionService(ModelingSubmissionRepository modelingSubmission
+        }
+
         if (modelingSubmission.isSubmitted()) {
             submit(modelingSubmission, modelingExercise);
+        } else if (modelingExercise.getDueDate() != null && !modelingExercise.isEnded()) {
+            // save submission to HashMap if exercise not ended yet
+            AutomaticSubmissionService.updateSubmission(modelingExercise.getId(), user.getLogin(), modelingSubmission);
         }
 
         return result;
     }
 
+    /**
+     * Adds a model to compass service to include it in the automatic grading process.
+     *
+     * @param modelingSubmission    the submission which contains the model
+     * @param modelingExercise      the exercise the model belongs to
+     */
     public void submit(ModelingSubmission modelingSubmission, ModelingExercise modelingExercise) {
         this.compassService.addModel(modelingExercise.getId(), modelingSubmission.getId(), modelingSubmission.getModel());
     }
 
+    /**
+     * Find the model for given exerciseId, studentId and model id, which is the same as submission id.
+     *
+     * @param exerciseId    the exercise id for which to find the model
+     * @param studentId     the student id for which to find the model
+     * @param id            the model id
+     * @return the model JsonObject if found otherwise null
+     */
     public JsonObject getModel(long exerciseId, long studentId, long id) {
         if (jsonModelRepository.exists(exerciseId, studentId, id)) {
             return jsonModelRepository.readModel(exerciseId, studentId, id);
         }
         return null;
     }
 
-    private Result initializeResult(Participation participation, ModelingSubmission modelingSubmission) {
-        Result result = new Result().participation(participation).submission(modelingSubmission);
+    /**
+     * Initialize the attributes rated, successful and completionDate for a result or create a new one.
+     *
+     * @param participation     the participation the result should belong to
+     * @param result            null if new result otherwise the result for which to set the initial attributes
+     * @return the result with initialized attributes
+     */
+    private Result initializeResult(Participation participation, Result result) {
+        if (result == null) {
+            result = new Result().participation(participation);
+        }
         result.setRated(false);
         result.setSuccessful(false);
         result.setCompletionDate(ZonedDateTime.now());
-
-        participation.addResult(result);
-        participationService.save(participation);
-
         return result;
     }
 
+    /**
+     * Find the modelingSubmission by a given participation.
+     *
+     * @param participation    the participation for which to find the modelingSubmission
+     * @return the modelingSubmission if found otherwise null
+     */
     public ModelingSubmission findByParticipation(Participation participation) {
```

Comment:
rename method to findModelSubmissionByParticipation

Reply:
what happens if a participation has multiple modeling submissions?

Reply:
also the comment is somehow obvious, please describe how you handle different cases in the comment as well